preverifier.js

```diff
@@ -1,58 +1,43 @@
 /* This Source Code Form is subject to the terms of the Mozilla Public
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
-var jws = require('jws')
-var b64urltohex = require('browserid-crypto/lib/utils').b64urltohex
+
 var P = require('./promise')
+var JWTool = require('fxa-jwtool')
+
+module.exports = function (error, config) {
 
-module.exports = function (jwks, error, config) {
+  var jwtool = new JWTool(config.trustedJKUs)
 
   function nowSeconds() {
     return Math.floor(Date.now() / 1000)
   }
 
-  function parseJwt(str) {
-    try { return JSON.parse(Buffer(str, 'base64')) } catch (e) { return {} }
-  }
-
-  function jwtError(err, result, email, jwt) {
-    if (err) {
-      return { internal: err.message }
+  function jwtError(email, payload) {
+    if (payload.exp < nowSeconds()) {
+      return { exp: payload.exp }
     }
-    if (!result) {
-      return { sig: 'invalid' }
+    if (payload.aud !== config.domain) {
+      return { aud: payload.aud }
```

Comment:
I wonder if we should move some of these checks into fxa-jwtool to reduce duplication.  At least the `exp` and `aud` checks seem pretty generic.